resip/dum/ServerRegistration.cxx

```diff
@@ -143,6 +143,33 @@ ServerRegistration::accept(int statusCode)
    accept(success);
 }
 
+void
+ServerRegistration::reject(SipMessage& failure)
+{
+   failure.remove(h_Contacts);
+
+   InfoLog( << "rejected a registration " << mAor << " with statusCode=" << failure.header(h_StatusLine).statusCode() );
+
+   // First, we roll back the contact database to
+   // the state it was before the registration request.
+
+   // Async processing hasn't actually updated the database yet, so no need to roll back.
```

Comment:
There is bunch of code duplication here with the existing reject method.  Would be nice to consolidate these two methods more.  For example the existing Reject(int statusCode) method could call this new one after it has created the SipMessage to send.  Similar to the way the 2 accept methods work, something like:

void
ServerRegistration::reject(int statusCode)
{
   SipMessage failure;
   mDum.makeResponse(failure, mRequest, statusCode);
   reject(failure);
}

Reply:
Agreed. I've updated reject(int) to call reject(SipMessage&).